gensim/models/word2vec.py

```diff
@@ -1216,12 +1216,20 @@ def prepare_vocab(self, hs, negative, wv, update=False, keep_raw_vocab=False, tr
             sorted_vocab = sorted(sorted_vocab_list, key=lambda word: word[1], reverse=True)
 
             if self.max_vocab < len(sorted_vocab):
-                calc_min_count = sorted_vocab[self.max_vocab][1]
+                if sorted_vocab[self.max_vocab][1] != sorted_vocab[self.max_vocab - 1][1]:
```

Comment:
There's no need for this check. a `calc_min_count = self.raw_vocab[sorted_vocab[self.max_vocab]]` will always set the threshold to the exact level necessary to eliminate the words at `max_vocab` and later ranks.

Reply:
I think you meant : `calc_min_count = self.raw_vocab[sorted_vocab[self.max_vocab]] + 1`